Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1107,16 +1296,34 @@ private void AddLinks()
                     ball.name = "midpoint_" + i.ToString();
                     ball.transform.localPosition = edgeCenters[i];
 
+                    // Align handle with its edge assuming that the prefab is initially aligned with the up direction 
+                    if (edgeAxes[i] == CardinalAxisType.X)
+                    {
+                        Quaternion realignment = Quaternion.FromToRotation(Vector3.up, Vector3.right);
+                        ball.transform.localRotation = realignment * ball.transform.localRotation;
+                    }
+                    else if (edgeAxes[i] == CardinalAxisType.Z)
+                    {
+                        Quaternion realignment = Quaternion.FromToRotation(Vector3.up, Vector3.forward);
+                        ball.transform.localRotation = realignment * ball.transform.localRotation;
+                    }
+
                     if (rotationHandlePrefabColliderType == RotationHandlePrefabCollider.Sphere)
                     {
-                        SphereCollider collider = ball.AddComponent<SphereCollider>();
-                        collider.radius = 0.5f * rotationHandleDiameter;
+                        SphereCollider sphereCollider = ball.AddComponent<SphereCollider>();
+                        sphereCollider.radius = 0.5f * rotationHandleDiameter;
+                        float localScale = ball.transform.localScale.x;
+                        localScale = localScale == 0.0f ? 0.0001f : localScale;
+                        sphereCollider.radius = 0.04f / localScale;
                     }
-                    else
+                    else if( rotationHandlePrefabColliderType == RotationHandlePrefabCollider.Box)
                     {
                         Debug.Assert(rotationHandlePrefabColliderType == RotationHandlePrefabCollider.Box);
                         BoxCollider collider = ball.AddComponent<BoxCollider>();
-                        collider.size = rotationHandleDiameter * Vector3.one;
+                        float localScale = ball.transform.localScale.x;
+                        localScale = localScale == 0.0f ? 0.01f : localScale;
+                        float finalScale = rotationHandleColliderDiameter * 0.02f;
```

Comment:
It woul be nice to avoid magic numbers like this. Can you explain why you have this multiplier here? Especially since you re-use 0.02f down below...

Reply:
yes

Reply:
Could you please provide an explanation? @cre8ivepark I was hoping that this magic number would be explained and moved out into a variable, but it's not something that I would block on. It would just be nice to reduce magic numbers in our code.